cedar-policy-core/src/ast/policy.rs

```diff
@@ -943,91 +992,221 @@ pub struct TemplateBody {
     non_scope_constraints: Arc<Expr>,
 }
 
+#[derive(Clone, Hash, Eq, PartialEq, Debug)]
+/// Policy datatype. This is used for both templates (in which case it contains
+/// slots) and static policies (in which case it contains zero slots).
+pub enum TemplateBody {
+    /// Represents a valid template body
+    TemplateBody(TemplateBodyImpl),
+    #[cfg(feature = "tolerant-ast")]
+    /// Represents a policy that failed to parse
+    TemplateBodyError(PolicyID),
+}
+
+impl Serialize for TemplateBody {
```

Comment:
After #1520 we no longer have `Serialize` for this type at all, and I don't see a reason to add back `Serialize` in this PR (see reasoning in #1520)

Reply:
Oh nice! I will rebase and remove